arkane/statmech.py

```diff
@@ -92,38 +95,117 @@ def load(self):
         Load the scan energies from the file. Returns arrays containing the
         angles (in radians) and energies (in J/mol).
         """
-        angles, energies = [], []
-        angle_units, energy_units, angle_factor, energy_factor = None, None, None, None
+        file_ext = pathlib.Path(self.path).suffix
+        print(file_ext.lower())
+        if file_ext.lower() == '.csv':
+            angle_unit, energy_unit, angles, energies = self.load_csv()
+        elif file_ext.lower() in ['.yml', '.yaml']:
+            angle_unit, energy_unit, angles, energies = self.load_yaml()
+        else:
+            angle_unit, energy_unit, angles, energies = self.load_text()
+
+        try:
+            angle_factor = self.angle_factors[angle_unit]
+        except KeyError:
+            raise ValueError(f'Invalid angle unit {angle_unit}.')
+        try:
+            energy_factor = self.energy_factors[energy_unit]
+        except KeyError:
+            raise ValueError(f'Invalid energy units {energy_unit}.')
+
+        angles = np.array(angles) / angle_factor
+        energies = np.array(energies) / energy_factor
+        energies -= energies[0]
 
+        return angles, energies
+
+    def load_csv(self):
+        """
+        Load scan energies from a CSV file. The CSV file should at least contain
+        two columns `'Angle (angle unit)'` and `'Energy (energy unit)'` with
+        corresponding values. For the headers, both 'units' should be replaced
+        by the units supported in `ScanLog.angle_factors` and `ScanLog.energy_factors`.
+        """
+        # Example format
+        # Angle (radians),Energy (kJ/mol)
+        # 0.000000,0.014725
+        # 0.174533,0.722311
+        df = pd.read_csv(self.path, header='infer')
+        # In case, the user's csv file has no header. Use the default header and units.
+        if df.columns[0][0].isnumeric():
+            df = pd.read_csv(self.path, names=['Angle (radians)', 'Energy (J/mol)'])
+            print(df.head())
+        for column in df.columns:
+            if 'angle' in column.lower():
+                try:
+                    angle_unit = column.split()[1][1:-1]
+                except IndexError:
+                    raise ValueError(f'Invalid headers ({column}) in the csv file.')
+                angles = df[column]
+            elif 'energy' in column.lower():
+                try:
+                    energy_unit = column.split()[1][1:-1]
+                except IndexError:
+                    raise ValueError(f'Invalid headers ({column}) in the csv file.')
+                energies = df[column]
+        return angle_unit, energy_unit, angles, energies
+
+    def load_yaml(self):
```

Comment:
Do we want to allow users to specify `pivots`, `top`, and `fit` directly in the YAML file?

Reply:
We could but, currently, this adds no benefit since we make these arguments required for `HinderedRotor`, as other file formats don't support reading the information from them. I decide to not add this into the scope of this PR.